File: server/app/graphql/mutations/profiles/create_profile.rb
module Mutations
  module Profiles
    class CreateProfile < SignInRequiredMutation
      field :profile, ObjectTypes::ProfileType, null: true

      argument :birthday, GraphQL::Types::ISO8601Date, required: true
      argument :gender, EnumTypes::Profile::ProfileGenderEnum, required: true
      argument :introduction, String, required: false
      argument :avatar_url, String, required: false

      def resolve(**args)
        profile = current_user.create_profile!(**args)

        { profile: }
      end
    end
  end
end

File: client/features/auth/api/createProfile.ts
import { gql, useMutation } from "@apollo/client";
import { CreateProfilePayload } from "@/graphql/graphql";

const CREATE_PROFILE = gql`
  mutation CreateProfile($birthday: ISO8601Date!, $gender: ProfileGenderEnum!, $introduction: String, $avatarUrl: String) {
    createProfile(input: { birthday: $birthday, gender: $gender, introduction: $introduction, avatarUrl: $avatarUrl }) {
      profile {
        id
      }
    }
  }
`;

export const useCreateProfile = () => {
  const [createProfile, { loading }] = useMutation<{ createProfile: CreateProfilePayload }>(CREATE_PROFILE);
  return { createProfile, loading };
};